### ISSUE TITLE
mpfr 4.0.0: deprecated functions

### ISSUE THREAD

**Issue by kkatrio** [CGAL/cgal#2790]
## Issue Details

There are warnings while compiling cgal when using latest mpfr version 4.0.0 regarding function `mpfr_root`. In their latest changes (http://www.mpfr.org/mpfr-current/#changes) they report that 

> Functions mpfr_grandom and mpfr_root are deprecated and will be removed in a future release.

```
/cgal/Number_types/include/CGAL/GMP/Gmpfr_type.h:889:49: warning: 
‘int mpfr_root(mpfr_ptr, mpfr_srcptr, long unsigned int, mpfr_rnd_t)’ is deprecated [-Wdeprecated-declarations]
         mpfr_root(result.fr(),fr(),k,_gmp_rnd(r));
```

## Environment

Non-rolling distros still have version 3.1.4-1 in their repositories.

**Comment by lrineau** [comment]
Only `mpfr_root` seems to be used in CGAL:
https://github.com/CGAL/cgal/blob/2e7c3ee5b8c8bd2b91539636d52607abc6c607f4/Number_types/include/CGAL/GMP/Gmpfr_type.h#L886-L901

It should be replaced by [`mpfr_rootn_ui`], according the [the changelog]:
>  New function `mpfr_rootn_ui`, similar to `mpfr_root`, but agreeing with the rootn function of the IEEE 754-2008 standard.

  [the changelog]: http://www.mpfr.org/mpfr-current/#changes
  [`mpfr_rootn_ui`]: http://www.mpfr.org/mpfr-current/mpfr.html#index-mpfr_005frootn_005fui

`mpfr_rootn_ui`, and `mpfr_root` have the same API, and the difference in behavior certainly does not matter for our usage.

**Comment by lrineau** [comment]
Konstantinos @kkatrio, could you propose a pull-request to fix that issue?

**Comment by kkatrio** [comment]
I have applied the fix locally: No warnings with the new function when using mpfr4. I would propose the PR, however, the new function does not exist in mpfr's version 3! What the hell.

**Comment by sloriot** [comment]
There should be a macro to check which version of mpfr is used. You can do the replacement conditionally then.
According to the [documentation](http://www.mpfr.org/mpfr-current/mpfr.html#MPFR-Interface) the macros are:
```
Macro: MPFR_VERSION
Macro: MPFR_VERSION_MAJOR
Macro: MPFR_VERSION_MINOR
Macro: MPFR_VERSION_PATCHLEVEL
Macro: MPFR_VERSION_STRING
```

**Comment by afabri** [comment]
@lrineau this is a low hanging fruit

**Comment by kkatrio** [comment]
@afabri What does _a low hanging fruit_ mean?

**Comment by lrineau** [comment]
Fixed by #2808 in both `master` and `releases/CGAL-4.11-branch`.
